newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorPropertiesEditor.js

```diff
@@ -124,10 +124,14 @@ export default class EventsBasedBehaviorPropertiesEditor extends React.Component
 
   _setChoiceExtraInfo = (property: gdNamedPropertyDescriptor) => {
     return (newExtraInfo: Array<string>) => {
+      const defaultValueIndex = getExtraInfoArray(property).indexOf(
+        property.getValue()
+      );
       const vectorString = new gd.VectorString();
       newExtraInfo.forEach(item => vectorString.push_back(item));
       property.setExtraInfo(vectorString);
       vectorString.delete();
+      property.setValue(newExtraInfo[defaultValueIndex]);
```

Comment:
Any chance this is defaultValueIndex can be -1? In which case, prefer to do
```suggestion
      property.setValue(newExtraInfo[defaultValueIndex] || "");
```

to avoid passing undefined.
For example, if newExtraInfo is empty.